eucadeploy/componentdeployer.py

```diff
@@ -23,9 +24,10 @@ def __init__(self, password, environment_file='environment.yml',
         with hide(*self.hidden_outputs):
             local('if [ ! -d eucalyptus-cookbook ]; then '
                   'git clone '
-                  'https://github.com/eucalyptus/eucalyptus-cookbook;'
-                  'fi')
+                  '{0} eucalyptus-cookbook;'
+                  'fi'.format(cookbook))
             local('cd eucalyptus-cookbook; git checkout {0};'.format(cookbook_branch))
+            local('git pull')
```

Comment:
Nice catch, I incorrectly assumed that changing dir in the previous execute local line would persist.